### PR TITLE
team-building mobile tab bar animation

### PR DESCRIPTION
**Pull request by mlsteele**
Mobile service tab bar animation. Desktop should be almost unchanged by this PR.

- [x] test android again
Will test android again tomorrow. Android perf for the animation is unacceptable. I'm hoping that's due to being a dev build.

@keybase/picnicsquad @keybase/react-hackers 

image
image
image

### CODE REVIEW

**Comment by mlsteele** [comment]
sorry, android sometimes retriggers its own animation by scrolling. Going to move the state down into ServiceTabBar and debounce or something

**Comment by mlsteele** [comment]
Ready again

File: shared/team-building/service-tab-bar.native.tsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
+import * as React from 'react'
+import * as Kb from '../common-adapters/mobile.native'
+import * as Styles from '../styles'
+import {useTimeout} from '../common-adapters/use-timers'
```
Comment (chrisnojima) [inline shared/team-building/service-tab-bar.native.tsx]: this is in Kb

File: shared/team-building/service-tab-bar.native.tsx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+  let n = Math.floor(p) + 0.5
+  if (p % 1 < 0.5) {
+    n -= 1
+  }
+  // n = count that will fit onscreen at returned size
+  return containerWidth / n
+}
+
+const ServiceIcon = (props: IconProps) => {
+  const smallWidth = serviceMinWidthWhenSmall(Kb.NativeDimensions.get('window').width)
```
Comment (chrisnojima) [inline shared/team-building/service-tab-bar.native.tsx]: this is `Styles.dimensionWidth`

File: shared/team-building/service-tab-bar.native.tsx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+          props.isActive && {backgroundColor: serviceIdToAccentColor(props.service)},
+        ])}
+      />
+    </Kb.ClickableBox>
+  )
+}
+
+const undefToNull = (n: number | undefined | null): number | null => (n === undefined ? null : n)
+
+const deriveOnScroll = memoize((deferClose, locked, setShowLabels) => () => {
```
Comment (chrisnojima) [inline shared/team-building/service-tab-bar.native.tsx]: if you're memoizing a callback use `useCallback`

File: shared/team-building/service-tab-bar.native.tsx [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+              ))}
+            </Kb.ScrollView>
+          </Kb.Box2>
+        )}
+      </Kb.Animated>
+    </Kb.Box2>
+  )
+}
+
+const styles = Styles.styleSheetCreate({
```
Comment (chrisnojima) [inline shared/team-building/service-tab-bar.native.tsx]: use a closure now

File: shared/team-building/service-tab-bar.native.tsx [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+    height: 48,
+    position: 'relative',
+  },
+  inactiveTabBar: {
+    borderBottomWidth: 1,
+    borderColor: Styles.globalColors.black_10,
+    height: 2,
+  },
+  pendingIcon: {height: 17, width: 17},
+  resultCount: {},
```
Comment (chrisnojima) [inline shared/team-building/service-tab-bar.native.tsx]: 🔥

File: shared/team-building/service-tab-bar.native.tsx [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+  serviceIcon: {},
+  serviceIconContainer: {
+    flex: 1,
+    paddingBottom: Styles.globalMargins.tiny,
+    paddingTop: Styles.globalMargins.tiny,
+  },
+  tabBarContainer: {
+    backgroundColor: 'white',
```
Comment (chrisnojima) [inline shared/team-building/service-tab-bar.native.tsx]: no raw colors ever

File: shared/team-building/service-tab-bar.native.tsx (hunk 3 above)
Comment (mlsteele) [reply shared/team-building/service-tab-bar.native.tsx]: Thanks. I'm unsure when to memoize / useCallback or not. Should onClose here also useCallback? Should I learn how to profile, like would there be enough signal in this function?

Comment (chrisnojima) [reply shared/team-building/service-tab-bar.native.tsx]: you should def use the profiler. see: https://react-devtools-tutorial.now.sh/ for a nice tutorial
i'd lean on useCallback as its built for that purpose and its native to react

File: shared/team-building/service-tab-bar.desktop.tsx [hunk 7]
```diff
@@ -128,23 +80,16 @@ const styles = Styles.styleSheetCreate({
...
+    borderBottomWidth: 1,
+    borderColor: Styles.globalColors.black_10,
+    height: 2,
   },
-  pendingIcon: Styles.platformStyles({
-    isElectron: {height: 10, width: 10},
-    isMobile: {height: 17, width: 17},
-  }),
+  pendingIcon: {height: 10, width: 10},
   resultCount: {},
```
Comment (nathunsmitty) [inline shared/team-building/service-tab-bar.desktop.tsx]: do we need this empty object for anything?

Comment (mlsteele) [reply shared/team-building/service-tab-bar.desktop.tsx]: We don't. Will kill it in the next PR

**Pull request merged**